### PR TITLE
Name refactoring for tests, growth, and median

### PR DESCRIPTION
**Pull request by nathanvaughan-NOAA**
# What is the feature?

* Refactoring test names to remove _ and match google style recommendations
* Refactoring population to use weight_at_age[] vector instead of growth evaluate function
* Refactoring to replace the median parameter with inflection_point to be more descriptive and avoid overlap with R function names

# How have you implemented the solution?

We have refactored the code base to change names as described.

# Does the PR impact any other area of the project?

No this should not impact the rest of the code

# How to test this change

# Developer pre-PR checklist

- [x] I relied on GitHub actions to :test_tube: things for me while I sat on the :couch_and_lamp:.

### CODE REVIEW

**Comment by nathanvaughan-NOAA** [comment]
We know we are a couple of commits behind main but given the large number of open pull requests we were leaving another rebase until this is ready to be merged into main.

File: inst/include/population_dynamics/population/population.hpp
Comment (Bai-Li-NOAA) [inline inst/include/population_dynamics/population/population.hpp:176]: question[non-blocking]: do we want to use `this->weight_at_age[ages[age]]` or `this->weight_at_age[age]` throughout the codebase?

**Review by Bai-Li-NOAA — COMMENTED** [review]
This looks great! Thanks for the name refactoring. I can help with merging these changes into the main branch once the conflicts are resolved.

**Review by Bai-Li-NOAA — APPROVED** [review]

File: inst/include/population_dynamics/population/population.hpp
Comment (nathanvaughan-NOAA) [reply inst/include/population_dynamics/population/population.hpp:176]: We discussed this when we were writing and concluded that this->weight_at_age[age] will always be the expected index value while this->weight_at_age[ages[age]] would crash if at some point we allow no integer ages. At the moment they are both equivalent though.

**Comment by nathanvaughan-NOAA** [comment]
@ChristineStawitz-NOAA Looks like we already found a scenario where the merge block is an issue. This pull request says it cannot be rebased due to conflicts and it will now only let me choose Squash and merge which I don't want to do because these are 4 different issues that would be best kept separate for future interpretability.

**Comment by ChristineStawitz-NOAA** [comment]
@nathanvaughan-NOAA I came to the same conclusion once I tried to approve :) I removed that setting

**Comment by nathanvaughan-NOAA** [comment]
Hahaha sounds good, I'll merge once I fix these merge conflicts :)

**Pull request merged**
